particles/Services/particles/js/ImageClassifier.js

```diff
@@ -0,0 +1,108 @@
+/**
+ * @license
+ * Copyright 2019 Google LLC.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * Code distributed by Google as part of this project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+'use strict';
+
+defineParticle(({DomParticle, log, html, resolver}) => {
+
+  const template_ = html`
+<div>
+  <h2>Classification with a generic image classifier</h2>
+  <img style="max-width: 240px;" src="{{imageUrl}}" alt="Image to classify"><br>
+  <div id="progress-bar">{{progress}}</div>
+  <div>
+    <div>Label: </span><span>{{label}}</div>
+    <div>Confidence: </span><span>{{probability}}</div>
+  </div>
+</div>
+  `;
+
+  const url = resolver(`ImageClassifier/../../assets/waltbird.jpg`);
+  const modelUrl = 'https://tfhub.dev/google/imagenet/mobilenet_v1_100_224/classification/1';
+  const labelUrl = resolver(`ImageClassifier/../../assets/ImageNetLabels.txt`);
+
+  return class extends DomParticle {
+    get template() {
+      return template_;
+    }
+
+    update({}, state) {
+      // TODO(sjmiles): update() is called during SpecEx, while
+      // render() is not. We'll put our processing code in render()
+      // to avoid being expensive at SpecEx time.
+    }
+
+    render({}, state) {
+      // formerly update
+      if (!state.loaded) {
+        state.loaded = true;
+        this.loadModel(modelUrl, {fromTFHub: true});
+      }
+      this.getModelLabels();
```

Comment:
Same as above -- async but we're not awaiting, is that ok? Probably, because `getModelLabels` calls `setState` which will cause a re-render, but would be good to hear @sjmiles thought.